common/src/main/java/org/wso2/testgrid/common/infrastructure/InfrastructureCombination.java

```diff
@@ -47,6 +47,17 @@ public InfrastructureCombination(Set<InfrastructureParameter> parameters) {
         this.parameters.addAll(parameters);
     }
 
+
+    /**
+     * Initializes an @{@link InfrastructureCombination} object with the given
+     * single infrastructure parameter.
+     *
+     * @param parameter infrastructure parameter.
+     */
+    public InfrastructureCombination(InfrastructureParameter parameter) {
```

Comment:
I'm not sure accepting just one InfrastructureParameter by InfrastructureCombination class is good. How about we change this to accept a var-args instead (`public InfrastructureCombination(InfrastructureParameter parameter...) {
`)?

Alternatively, we can also introduce a method like #addInfrastructureParameter instead.

Reply:
+1